### DISCUSSION TITLE
[Question]: librechat.yaml - Version field efficacy

### DISCUSSION THREAD

**Discussion by tip-dteller** [danny-avila/LibreChat#2892] [Q&A] [answered]
### What is your question?

When new versions are released whether Release or Dev versions, there's a noticeable affect on the yaml structure.
The documentation states the the "version field" is 1.0.8.
yet when using Dev versions, the logs print out "latest version is 1.1.2"

why am i raising this?
because when using v0.7.2 with librechat.yaml version 1.0.5/7/8/9 - the STT\TTS fields are thrown with an error "unrecognized keys".

@danny-avila could you please clarify the usage and when to update the Version field? 
as it appears to go through many changes.

### More Details

```
librechat-ops-67dfd48cbb-2rr8p librechat-chart 2024-05-28 11:37:39 info:
librechat-ops-67dfd48cbb-2rr8p librechat-chart Outdated Config version: 1.0.9
librechat-ops-67dfd48cbb-2rr8p librechat-chart Latest version: 1.1.2
```

Here its 0.7.2 with yaml version 1.0.8
the stt and tts are as specified by the docs.

```
    version: 1.0.8
    cache: true
    stt:
      openai:
        apiKey: '${LITELLM_OPENAI_API_KEY}'
        url: 'http://litellm-chatops.librechat:4000/v1/audio/transcriptions'
        model: 'whisper-1'
```

```
librechat-ops-58f9f694f9-zlnw2 librechat-chart > LibreChat@0.7.2 backend
librechat-ops-58f9f694f9-zlnw2 librechat-chart > cross-env NODE_ENV=production node api/server/index.js
...
librechat-ops-58f9f694f9-zlnw2 librechat-chart 2024-05-28 11:32:49 error: Invalid custom config file at /app/librechat.yaml [
librechat-ops-58f9f694f9-zlnw2 librechat-chart   {
librechat-ops-58f9f694f9-zlnw2 librechat-chart     "code": "unrecognized_keys",
librechat-ops-58f9f694f9-zlnw2 librechat-chart     "keys": [
librechat-ops-58f9f694f9-zlnw2 librechat-chart       "stt",
librechat-ops-58f9f694f9-zlnw2 librechat-chart       "tts"
librechat-ops-58f9f694f9-zlnw2 librechat-chart     ],
librechat-ops-58f9f694f9-zlnw2 librechat-chart     "path": [],
librechat-ops-58f9f694f9-zlnw2 librechat-chart     "message": "Unrecognized key(s) in object: 'stt', 'tts'"
librechat-ops-58f9f694f9-zlnw2 librechat-chart   }
librechat-ops-58f9f694f9-zlnw2 librechat-chart ]
````

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Hi @tip-dteller 

First, let me clarify that when using version-specific images, you should be expecting errors like this.

v0.7.2 is every commit up to commit https://github.com/danny-avila/LibreChat/commit/4a32d7466aaaea5ba1fbe8b8e38ef115796166c0

Release info: https://github.com/danny-avila/LibreChat/releases/tag/v0.7.2

This does not include the latest changes for text-to-speech, and speech-to-text. In order for those fields to be recognized, you would need to use one of the latest of these images:
- https://github.com/users/danny-avila/packages/container/package/librechat-dev
- https://github.com/users/danny-avila/packages/container/package/librechat-dev-api

The version field is simply to denote changes to the YAML structure. You should update it when you are on the latest build/tag, and match your YAML accordingly, so then you can be aware when there are changes.

I understand part of the confusion comes from the docs being updated. This is because a large number of users try out the latest build, but we could be specific about versioning in the features there.

The latest release is v0.7.3-rc, which itself doesn't include stt/tts either.

We are due another release, likely v0.7.3-rc2 because my goals for v0.7.3 are not yet complete, which would include the most stable up to this point.

**Reply by apeironian** [reply]
Thanks for this, but I still have a question:

I've installed the full 0.7.3 release and configured the stt and tts fields in librechat.yaml per the documentation, but I'm still getting the same error as above (unrecognized keys). Do I need to manually update the version identifier in the yaml? 

Thanks for your help.

**Comment by tip-dteller** [comment]
I understand, its difficult to align between the "production version" and the "development version" in the documentation :)
